exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceStreamObservers.java

```diff
@@ -0,0 +1,206 @@
+      new StreamObserver<ExportTraceServiceResponse>() {
+        @Override
+        public void onNext(ExportTraceServiceResponse value) {
+          // Do nothing since ExportTraceServiceResponse is an empty message.
+        }
+
+        @Override
+        public void onError(Throwable t) {
+          // If there's an error, reset the reference to exportRequestObserver to force
+          // re-establish a connection.
+          // TODO(songya): add retries on this connection before starting a new connection.
+          logger.log(Level.WARNING, "Export stream is disconnected because of: {0}", t);
+          /*@Nullable*/ StreamObserver<ExportTraceServiceRequest> exportRequestObserver =
+              exportRequestObserverRef.get();
+          if (exportRequestObserver != null) {
+            exportRequestObserver.onError(t); // Tries to notify OC-Agent about the error.
+            resetExportClient();
+          }
+        }
+
+        @Override
+        public void onCompleted() {
+          resetExportClient();
+        }
+      };
+
+  // Stream observer for UpdatedLibraryConfig messages. Once there's a new UpdatedLibraryConfig,
+  // apply it to the global TraceConfig.
+  static final StreamObserver<UpdatedLibraryConfig> updatedLibraryConfigObserver =
+      new UpdatedLibraryConfigObserver(Tracing.getTraceConfig());
+
+  @VisibleForTesting
+  static class UpdatedLibraryConfigObserver implements StreamObserver<UpdatedLibraryConfig> {
+
+    // Inject a mock TraceConfig in unit tests.
+    private final TraceConfig traceConfig;
+
+    UpdatedLibraryConfigObserver(TraceConfig traceConfig) {
+      this.traceConfig = traceConfig;
+    }
+
+    @Override
+    public void onNext(UpdatedLibraryConfig value) {
+      TraceParams updatedTraceParams = TraceProtoUtils.getUpdatedTraceParams(value, traceConfig);
+      traceConfig.updateActiveTraceParams(updatedTraceParams);
+
+      /*@Nullable*/ StreamObserver<CurrentLibraryConfig> currentConfigObserver =
+          currentConfigObserverRef.get();
+      // Add a check in case that connection dropped while config is being updated.
+      if (currentConfigObserver != null) {
+        // Bouncing back CurrentLibraryConfig to Agent.
+        io.opencensus.proto.trace.v1.TraceConfig currentTraceConfigProto =
+            TraceProtoUtils.getCurrentTraceConfig(traceConfig);
+        CurrentLibraryConfig currentLibraryConfig =
+            CurrentLibraryConfig.newBuilder().setConfig(currentTraceConfigProto).build();
+        currentConfigObserver.onNext(currentLibraryConfig);
+      }
+    }
+
+    @Override
+    public void onError(Throwable t) {
+      // If there's an error, reset the reference to currentConfigObserver to force
+      // re-establish a connection.
+      // TODO(songya): add retries on this connection before starting a new connection.
+      logger.log(Level.WARNING, "Config stream is disconnected because of: {0}", t);
```

Comment:
Better to use this:
https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)

logger.log(Level.WARNING, "Config stream is disconnected.", t);

Reply:
Done.